src/foam/nanos/pm/PM.js

```diff
@@ -128,25 +167,33 @@ foam.CLASS({
   }
   */
 
-  public PM(ClassInfo clsInfo, String name) {
-    setName(name);
+  public PM(ClassInfo clsInfo, String... name) {
+    setName(combine(name));
     setClassType(clsInfo);
     init_();
   }
 
-  public PM(Class cls, String name) {
-    setName(name);
+  public PM(Class cls, String... name) {
+    setName(combine(name));
     foam.core.ClassInfoImpl clsInfo = new foam.core.ClassInfoImpl();
     clsInfo.setObjClass(cls);
     clsInfo.setId(cls.getName());
     setClassType(clsInfo);
     init_();
   }
 
-  public PM(FObject fo, String name) {
+  public PM(FObject fo, String... name) {
     this(fo.getClassInfo(), name);
   }
 
+  private static String combine(String... args) {
```

Comment:
Create another constructor with just (Object... args) 
take first arg as id, and as mentioned above, look for exception.

Reply:
So the second instruction here was wrong, sorry.

Reply:
sorry, ignore all of this.